### PR TITLE
Prison Population Over Time chart

### LINKED ISSUE #1291: Add population over time chart to the prison page

### PR DESCRIPTION
**Pull request by jovergaag**
## Description of the change

Adds the Prison Population Over Time chart, and associated notes, filters, and styling.

Also cleans up the projections summaries metric/component since we will be using the "temp" solution as the long term solution moving forward, and no longer need the summaries metric.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #1291

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by jovergaag** [comment]
@Catacola This is ready for review! I'm still working on a little issue with the dashed line spacing in the projection but its all ready other than that.

File: src/core/PopulationSummaryMetrics/index.tsx [hunk 1]
```diff
@@ -14,75 +14,204 @@
...
-  // TODO: add in Error state
   if (isError) {
     return null;
   }
 
   if (isLoading) {
     return (
       <div className="PopulationSummaryMetrics">
-        <HistoricalSummaryMetrics isLoading />
-        <ProjectedSummaryMetrics isLoading />
```
Comment (Catacola) [inline src/core/PopulationSummaryMetrics/index.tsx:68]: What was the motivation behind removing these components and collapsing everything into one file?

File: src/core/models/PathwaysMetric.ts [hunk 2]
```diff
@@ -41,12 +43,16 @@ export type BaseMetricConstructorOptions<RecordFormat extends MetricRecord> = {
  * The preferred way to instantiate `Metric` subclasses is in the
  * MetricsStore.
  */
-export default abstract class PathwaysMetric<RecordFormat extends MetricRecord>
+export default abstract class Metric<RecordFormat extends MetricRecord>
```
Comment (Catacola) [inline src/core/models/PathwaysMetric.ts:46]:
```suggestion
export default abstract class PathwayMetric<RecordFormat extends MetricRecord>
```

**Review by Catacola — COMMENTED** [review]
This is looking pretty good. I like the new indigo line. I noticed that the line color changed for the existing projection plot as well. This is so minor that it's probably fine, but I just wanted to flag it.

File: src/core/models/PathwaysMetric.ts (hunk 2 above)
Comment (jovergaag) [reply src/core/models/PathwaysMetric.ts:46]: Ugh. Apparently it was not a clean merge. Thanks for finding this again.

File: src/core/PopulationSummaryMetrics/index.tsx (hunk 1 above)
Comment (jovergaag) [reply src/core/PopulationSummaryMetrics/index.tsx:68]: These two `HistoricalSummaryMetrics` and `ProjectedSummaryMetrics` were two of the unused components while we were using `TempPopulationSummaryMetrics`. I removed all of the unused components, and moved the temp component to be the permanent component without changing any of the styling or component hierarchy. I'll look into bringing these two back to life and working them into the temp/now permanent component structure.

**Comment by jovergaag** [comment]
> This is looking pretty good. I like the new indigo line. I noticed that the line color changed for the existing projection plot as well. This is so minor that it's probably fine, but I just wanted to flag it.

You're correct, @soniachokshi said that it was fine to change the line on the existing plot as well.

File: src/core/PopulationSummaryMetrics/index.tsx (hunk 1 above)
Comment (jovergaag) [reply src/core/PopulationSummaryMetrics/index.tsx:68]: I tried to clean up the `PopulationSummaryMetrics` component a little more. It is now using `MetricsCard` and I moved the styling into the scss file. LMK what you think.

File: src/core/PopulationTimeSeriesChart/PopulationTimeSeriesChart.scss [hunk 3]
```diff
@@ -41,7 +41,7 @@
   }
 
   &__ProjectedLine {
-    stroke-dasharray: 5 7;
+    stroke-dasharray: 7 9;
```
Comment (Catacola) [inline src/core/PopulationTimeSeriesChart/PopulationTimeSeriesChart.scss:44]: This is one of those weird things we're just going to need to play with over time until we reimplement the projected line. The wonky behavior comes from the fact that it's not one line, but two: one going there and one going back. This then causes interference between the dashes and leads to odd behavior.

File: src/core/PopulationSummaryMetrics/index.tsx [hunk 4]
```diff
@@ -14,21 +14,17 @@
 // You should have received a copy of the GNU General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 // =============================================================================
-// TODO(recidiviz-data/issues/8751): Remove this component when
-// projections data is stable and auto exported
-import "./LoadingMetrics.scss";
 import "./PopulationSummaryMetrics.scss";
 
-import { Card, CardSection } from "@recidiviz/design-system";
+import cn from "classnames";
```
Comment (Catacola) [inline src/core/PopulationSummaryMetrics/index.tsx:19]: Flagging what github flagged.

**Review by Catacola — APPROVED** [review]
Thanks for the cleanup!

File: src/core/PopulationTimeSeriesChart/PopulationTimeSeriesChart.scss (hunk 3 above)
Comment (jovergaag) [reply src/core/PopulationTimeSeriesChart/PopulationTimeSeriesChart.scss:44]: Oh is that whats happening? I couldn't quite figure it out.

**Pull request merged**